### PR TITLE
Added API documentation and script for doc generation

### PR DESCRIPTION
**Pull request by gokulmInfy**
Ready for Review

### CODE REVIEW

File: src/M2XStreamClient.h [hunk 1]
```diff
@@ -60,41 +60,47 @@ class M2XStreamClient {
...
-  // +at+ - ISO 8601 timestamp for the set of values
-  // +values+ - Values to post, the length of the array should
-  // be exactly +streamNum+
+/**
+  * Method for <a href="https://m2x.att.com/developer/documentation/v2/device#Post-Device-Update--Single-Values-to-Multiple-Streams-">Post Device Update</a> endpoint.
+  * Post multiple values of a single device at once.
+  * @param deviceId - id of the device to post values
+  * @param streamNum - Number of streams to post
+  * @param names - Array of stream names, the length of the array should  be exactly +streamNum+
+  * @param values - Array of values to post, the length of the array should  be exactly +streamNum+. Notice that the array of +values+ should
```
Comment (amclain) [inline src/M2XStreamClient.h]: Long lines should be split at 80 columns or less so they can be easily read in a text editor. URLs will unfortunately run over sometimes, but the following text should be split as soon as reasonable.

Also, be sure to remove any double spaces between words caused by joining lines together, like in `should··be`.

Example:

image

File: src/M2XStreamClient.h [hunk 2]
```diff
@@ -60,41 +60,47 @@ class M2XStreamClient {
...
-  // first +counts[0]+ number of items contain values to post to the first
-  // stream, the succeeding +counts[1]+ number of items contain values
-  // for the second stream, etc. The length of this array should be
-  // the sum of all values in +counts+ array.
+  /**
+   * Method for <a href="https://m2x.att.com/developer/documentation/v2/device#Post-Device-Updates--Multiple-Values-to-Multiple-Streams-">Post Device Updates (Multiple Values to Multiple Streams)</a> endpoint.
+   * Post multiple values to M2X all at once.
+   * @param deviceId - id of the device to post values
+   * @param streamNum - Number of streams to post
+   * @param names - Array of stream names, the length of the array should be exactly +streamNum+
```
Comment (amclain) [inline src/M2XStreamClient.h]: References like `+streamNum+` should be converted to `` `streamNum` `` (backticks) so doxygen formats them correctly.

File: src/M2XStreamClient.h [hunk 3]
```diff
@@ -164,32 +176,52 @@ class M2XStreamClient {
...
+
+  /**
+   * Waits for a certain string pattern in the HTTP header, and returns
+   * once the pattern is found. In the pattern, you can use '*' to denote
+   * any character
+   */
   int waitForString(const char* str);
-  // Closes the connection
+
+  /** Closes the connection
```
Comment (amclain) [inline src/M2XStreamClient.h]: `/**` should be the only thing on this line; the text should be on a `*` line below it.

File: src/M2XStreamClient.h [hunk 4]
```diff
@@ -60,41 +60,47 @@ class M2XStreamClient {
                   const char* host = kDefaultM2XHost,
                   int port = kDefaultM2XPort);
 
-  // Push data stream value using PUT request, returns the HTTP status code
+/**
+ * Method for <a href="https://m2x.att.com/developer/documentation/v2/device#Update-Data-Stream-Value">Update Data Stream Value</a> endpoint.
+ * Push data stream value using PUT request, returns the HTTP status code
+ * @param deviceId - id of the device to post values
+ * @param streamName - Name of the stream.
+ * @param value - the value.<p>
```
Comment (amclain) [inline src/M2XStreamClient.h]: There are a few `<p>` tags in places that don't appear to do anything. These can be removed.

**Comment by gokulmInfy** [comment]
@amclain Ready for Review.

**Comment by amclain** [comment]
Looks good. Thanks!

**Pull request merged**